app/src/main/java/org/sagebionetworks/research/mpower/MainFragment.kt

```diff
@@ -78,13 +87,47 @@ class MainFragment : DaggerFragment() {
     @Inject
     lateinit var taskLauncher: TaskLauncher
 
+    @Inject
+    lateinit var reportRepo: ReportRepository
+
+    @Inject
+    lateinit var appConfigRepo: AppConfigRepository
+
+    lateinit var profileManager: ProfileManager
+
+    private var debounceCheckPassiveGait = Debounce(500, this::checkPassiveGait)
+
+    private lateinit var profileDataLoader: ProfileDataLoader
+
+    private val passiveDataAllowedObserver = Observer<ProfileDataLoader> { profileDataLoader ->
+        this.profileDataLoader = profileDataLoader
+        debounceCheckPassiveGait.schedule()
+    }
+
+    private fun checkPassiveGait() {
+        val passiveDataAllowed = profileDataLoader.getValueString("passiveDataAllowed")

```

Comment:
Reach out to Darryl and be able to comment why we need the Debouncer here.

Reply:
Reviewed with Darryl and we debounce here because the `ProfileDataLoader` observer fires multiple times as it is loading.